### PR TITLE
Use micromamba for pre-commit workflow

### PR DESCRIPTION
**Pull request by IzerOnadimQC**
Use `micromamba` instead of `mamba` for pre-commit.

### CODE REVIEW

File: .github/workflows/ci-pre-commit.yml [hunk 1]
```diff
@@ -4,8 +4,21 @@ on: [push, pull_request]
...
 jobs:
   build:
+    name: "Linux - pre-commit checks - Python 3.10"
+    timeout-minutes: 30
     runs-on: ubuntu-latest
+    env:
+      PRE_COMMIT_USE_MICROMAMBA: 1
     steps:
+    - name: Checkout branch
     - uses: actions/checkout@v3
```
Comment (IzerOnadimQC) [inline .github/workflows/ci-pre-commit.yml:14]: `quantcore.thek` uses `actions/checkout@v4` - I looked at the changes and there doesn’t seem to be anything too major, should I use the same here?

File: .github/workflows/ci-pre-commit.yml [hunk 2]
```diff
@@ -4,8 +4,21 @@ on: [push, pull_request]
...
-    - uses: actions/setup-python@v4
+    - name: Set up micromamba
+      uses: mamba-org/setup-micromamba@2b72821d5ad7f6da3c003a3684ce341bf187b46f
+    - name: Add micromamba to GITHUB_PATH
+      run: echo "${HOME}/micromamba-bin" >> "$GITHUB_PATH"
+    - name: Install Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.10"
```
Comment (IzerOnadimQC) [inline .github/workflows/ci-pre-commit.yml:22]: Other parts of the `plateau` CI use python 3.11 - any reason I shouldn’t update this to do the same?

File: .github/workflows/ci-pre-commit.yml (hunk 1 above)
Comment (jtilly) [reply .github/workflows/ci-pre-commit.yml]: I would stick to v3 and be consistent across actions in plateau, see https://github.com/search?q=repo%3Adata-engineering-collective%2Fplateau%20checkout&type=code.

At a later point, we can add dependabot to update actions.

Comment (IzerOnadimQC) [reply .github/workflows/ci-pre-commit.yml]: Ok - leaving it as is

File: .github/workflows/ci-pre-commit.yml (hunk 2 above)
Comment (jtilly) [reply .github/workflows/ci-pre-commit.yml]: The version here really doesn't matter (feel free to use whatever feels most consistent). It's just the version for `pre-commit`. Each hook runs in a separate environment anyways.

**Review by jtilly — APPROVED** [review]
> succeeded 42 minutes ago in 2m 27s

🚀

File: .github/workflows/ci-pre-commit.yml (hunk 2 above)
Comment (IzerOnadimQC) [reply .github/workflows/ci-pre-commit.yml]: I just checked out the `setup-python` documentation at it seems in their docs they're still using 3.10. I doubt there's any significance to that beyond not having updated their docs, but I'll leave it as is for now

**Pull request merged**